src/zenml/pipelines/base_pipeline.py

```diff
@@ -136,23 +134,50 @@ def __init__(self, *args: BaseStep, **kwargs: Any) -> None:
         """
         kwargs.update(getattr(self, INSTANCE_CONFIGURATION))
         self.enable_cache = kwargs.pop(PARAM_ENABLE_CACHE, True)
-        self.required_integrations = kwargs.pop(PARAM_REQUIRED_INTEGRATIONS, ())
-        self.requirements_file = kwargs.pop(PARAM_REQUIREMENTS_FILE, None)
-        if self.requirements_file:
-            logger.warning(
-                "The `requirements_file` argument has been deprecated. Please "
-                "use `requirements` instead to pass in either a string path "
-                "to a file listing your 'requirements' or a list of the "
-                "individual requirements."
-            )
-        self._requirements = kwargs.pop(PARAM_REQUIREMENTS, None)
-        self.dockerignore_file = kwargs.pop(PARAM_DOCKERIGNORE_FILE, None)
         self.secrets = kwargs.pop(PARAM_SECRETS, [])
 
+        self.docker_configuration = (
+            kwargs.pop(PARAM_DOCKER_CONFIGURATION, None)
+            or DockerConfiguration()
+        )
+        self._migrate_to_docker_config(kwargs)
+
         self.name = self.__class__.__name__
         self.__steps: Dict[str, BaseStep] = {}
         self._verify_arguments(*args, **kwargs)
 
+    def _migrate_to_docker_config(self, kwargs: Dict[str, Any]) -> None:
```

Comment:
The docstring for the method is missing here.

Reply:
Good catch, I added it!